### DISCUSSION TITLE
problem in installing @adonisjs/mail

### DISCUSSION THREAD

**Discussion by AhmedHashem2104** [adonisjs/core#2925] [Help] [answered]
i get error in
import { MailConfig } from '@ioc:Adonis/Addons/Mail'
i tried to install it with the following steps in this link
https://docs.adonisjs.com/guides/mailer
but it does not work

**Comment by brandonsousa** [comment]
what is the error you are giving?

**Reply by AhmedHashem2104** [reply]
MailConfig is not exported member

**Reply by AhmedHashem2104** [reply]
{
  "message": "Cannot read property 'fullName' of undefined",
  "stack": "Error: Cannot read property 'fullName' of undefined\n    at anonymous (/home/hashem/Documents/Qodeex New Website V2/server/resources/views/emails/welcome.edge:1:0)\n    at Template.reThrow (/home/hashem/Documents/Qodeex New Website V2/server/node_modules/edge.js/build/src/Template/index.js:178:15)\n    at template (eval at wrapToFunction (/home/hashem/Documents/Qodeex New Website V2/server/node_modules/edge.js/build/src/Template/index.js:64:20), <anonymous>:22:10)\n    at Template.renderCompiled (/home/hashem/Documents/Qodeex New Website V2/server/node_modules/edge.js/build/src/Template/index.js:84:57)\n    at Template.render (/home/hashem/Documents/Qodeex New Website V2/server/node_modules/edge.js/build/src/Template/index.js:139:21)\n    at EdgeRenderer.render (/home/hashem/Documents/Qodeex New Website V2/server/node_modules/edge.js/build/src/Renderer/index.js:37:103)\n    at Edge.render (/home/hashem/Documents/Qodeex New Website V2/server/node_modules/edge.js/build/src/Edge/index.js:233:35)\n    at Mailer.setEmailContent (/home/hashem/Documents/Qodeex New Website V2/server/node_modules/@adonisjs/mail/build/src/Mail/Mailer.js:39:52)\n    at Mailer.sendCompiled (/home/hashem/Documents/Qodeex New Website V2/server/node_modules/@adonisjs/mail/build/src/Mail/Mailer.js:65:20)\n    at Mailer.send (/home/hashem/Documents/Qodeex New Website V2/server/node_modules/@adonisjs/mail/build/src/Mail/Mailer.js:88:21)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)",
  "code": "E_RUNTIME_EXCEPTION"
}
it gives me this error in insomnia

**Comment by RomainLanz** [comment, marked as answer]
Hey @AhmedHashem2104! 👋 

What is the `emails/welcome.edge` looks like?
It seems that you are doing `something.fullName` while `something` is undefined.

**Reply by WeyIin** [reply]
@RomainLanz  is correct. Since you have followed official documentation, I can see it's calling `fullName` in user object, whereas your user object @AhmedHashem2104  - might be undefined.
Your email template isn't set up correctly - or you are not passing the correct data in.
